vignettes/convert.qmd

```diff
@@ -0,0 +1,346 @@
+
+The choice between converting to `DuckDB` and `Parquet` should be made based on your needs:
+
+-   If you plan to download large amounts of data, we recommend `DuckDB`, as it is one big file and it is easier to update it completely if for example you were to first use only dates for 2020, and then later decided to add more dates from 2021. In this case it would be better to delete the database and create it from scratch.
+
+-   If you only want certain dates, do analysis on them and then add more dates later, `Parquet` may be better, as each day is saved in a separate file, just like the original CSV files. Therefore updating a folder of `Parquet` files is as easy as just creating a new file only for the missing date.
+
+
+In our testing (see @fig-csv-duckdb-parquet-speed), using `DuckDB` database has a significant speed advantage over running the analysis on raw CSV.gz files (`DuckDB` up to 8 times faster), or even on `Parquet` files (`DuckDB` up to 5 times faster). The approaches to working with both formats are exactly the same, just use `{dplyr}` package and its functions for selecting, grouping, filtering and summarizing the data.
+
+![Data processing speed comparison: DuckDB engine running on CSV.gz files vs DuckDB database vs a folder of Parquet files](media/duckdb-parquet-csv-speed-mean-hourly-v1.svg){#fig-csv-duckdb-parquet-speed}
+
+<details><summary>Testing query</summary>
+
+For reference, here is a simple query we used for speed comparison in @fig-csv-duckdb-parquet-speed:
+
+```r
+data |>
+  group_by(id_origin, id_destination, time_slot) |> 
+  summarise(mean_hourly_trips = mean(n_trips, na.rm = TRUE),
+    .groups = "drop")
+```
+
+</details>
+
+
+@fig-csv-duckdb-parquet-speed also shows that `DuckDB` format will also give you the best performance even on low-end systems with limited memory and number of processor cores, conditional on a fast SSD storage.
+
+
+# Getting a single day with `spod_get()` {#spod-get}
+
+You can get a single day's worth of data as an in-memory object with `spod_get()`:
+
+```{r}
+dates <- c("2024-03-01")
+d_1 <- spod_get(type = "od", zones = "distr", dates = dates)
+class(d_1)
+```
+
+The output should look like this:
+
+```
+# Source:   table<od_csv_clean_filtered> [?? x 19]
+# Database: DuckDB v1.0.0 [... 6.5.0-45-generic:R 4.4.1/:memory:]
+   date       time_slot id_origin id_destination distance activity_origin  
+   <date>         <int> <fct>     <fct>          <fct>    <fct>            
+ 1 2024-03-01        19 01009_AM  01001          0.5-2    frequent_activity
+ 2 2024-03-01        15 01002     01001          10-50    frequent_activity
+```
+
+Note that this is a lazily-evaluated in-memory object (note the `:memory:` in the database path). This means that the data is not loaded into memory until you call `collect()` on it. This is useful for quick exploration of the data, but not recommended for large datasets.
+
+## `DuckDB` {#duckdb}
```

Comment:
Another argument for a restructuring. Why is 2.1 under a heading that says "get in one day" when it actually takes more than one day?